src/LayerTree/LayerTree.jsx

```diff
@@ -0,0 +1,246 @@
+  }
+
+  /**
+   * Determines what to do with new props.
+   *
+   * @param {Object} nextProps The new props.
+   */
+  componentWillReceiveProps(nextProps) {
+    if (!isEqual(this.props.layerGroup, nextProps.layerGroup)) {
+      this.treeNodesFromLayerGroup(nextProps.layerGroup);
+      this.checkedKeysFromLayerGroup(nextProps.layerGroup);
+    }
+  }
+
+  /**
+   * Creates TreeNodes from a given layergroup and sets the treeNodes in the state.
+   *
+   * @param {ol.layer.Group} groupLayer A grouplayer.
+   */
+  treeNodesFromLayerGroup(groupLayer) {
+    const layerArray = groupLayer.getLayers().getArray();
+    const treeNodes = layerArray.map((layer) => {
+      return this.treeNodeFromLayer(layer);
+    });
+    this.setState({treeNodes});
+  }
+
+  /**
+   * Determines the checkedKeys from a given layerGroup. Iterates over its children
+   * and adds visibile layers to the checkedKeysArray in the state.
+   *
+   * @param {ol.layer.Group} groupLayer A grouplayer.
+   */
+  checkedKeysFromLayerGroup(groupLayer) {
+    const allLayers = MapUtils.getAllLayers(groupLayer);
+    let checkedKeys = [];
+    allLayers.forEach((layer) => {
+      if (layer.getVisible()) {
+        checkedKeys.push(layer.ol_uid.toString());
+      }
+    });
+    this.setState({checkedKeys});
+  }
+
+  /**
+   * Creates a treeNode from a given layer.
+   *
+   * @param {ol.layer.Layer} layer The given layer.
+   * @return {TreeNode} The corresponding TreeNode Element.
+   */
+  treeNodeFromLayer(layer) {
+    let childNodes;
+    let treeNode;
+    if (layer instanceof OlGroupLayer) {
+      const childLayers = layer.getLayers().getArray();
+      childNodes = childLayers.map((childLayer) => {
+        return this.treeNodeFromLayer(childLayer);
+      });
+    }
+    treeNode = <TreeNode
+      title={layer.get('name')}
+      key={layer.ol_uid}
+    >
+      {childNodes}
+    </TreeNode>;
+    return treeNode;
+  }
+
+  /**
+   * Sets the visibility of a layer due to its checked state.
+   *
+   * @param {Array<String>} checkedKeys Contains all checkedKeys.
+   * @param {e} checked The ant-tree event object for this event. See ant docs.
+   */
+  onCheck = (checkedKeys, e) => {
+    const {checked} = e;
+    const layer = MapUtils.getLayerByOlUid(this.props.map, e.node.props.eventKey);
+    if (layer) {
+      layer.setVisible(checked);
+    }
+    this.setState({checkedKeys});
+  }
+
+  /**
+   * The callback method for the drop event. Layers will get reordered in the map
+   * and the tree.
+   *
+   * @param {Object} e The ant-tree event object for this event. See ant docs.
+   */
+  onDrop = (e) => {
+    const dragLayer = MapUtils.getLayerByOlUid(this.props.map, e.dragNode.props.eventKey);
+    const dragInfo = this.getLayerPositionInfo(dragLayer);
+    const dropLayer = MapUtils.getLayerByOlUid(this.props.map, e.node.props.eventKey);
+    const dropPos = e.node.props.pos.split('-');
+    const location = e.dropPosition - Number(dropPos[dropPos.length - 1]);
```

Comment:
can `dropPos` be an empty array here?

Reply:
No it can't as split allways returns a array with minimial length of 1. This part of the code is copied from the ant examples so it should be quiet stable.